traffic_ops/traffic_ops_golang/staticdnsentry/staticdnsentry.go

```diff
@@ -81,8 +82,8 @@ func (staticDNSEntry *TOStaticDNSEntry) SetKeys(keys map[string]interface{}) {
 // Validate fulfills the api.Validator interface
 func (staticDNSEntry TOStaticDNSEntry) Validate() error {
 	errs := validation.Errors{
-		"host":    validation.Validate(staticDNSEntry.Host, validation.Required),
-		"address": validation.Validate(staticDNSEntry.Address, validation.Required),
+		"host":    validation.Validate(staticDNSEntry.Host, validation.Required, is.DNSName),
+		"address": validation.Validate(staticDNSEntry.Address, validation.Required, is.Host),
```

Comment:
To get 100% coverage I think we need to vary the validation based upon the type, i.e. 'A_RECORD' should be is.IPv4, 'AAAA_RECORD` should be is.IPv6, and 'CNAME_RECORD' should be is.DNSName.

Since I'm adding type validation in https://github.com/apache/trafficcontrol/pull/2572, I could also add those checks if you want.

Reply:
@rawlinp So you are centralizing *all* type validation?  If that's the case then sure and just let me know and I'll refer to it.

Reply:
Centralizing in the sense that everything that requires types from the types table can/will use the same shared function that also returns the type name when valid. So once that's in, you could just check the type name here and vary the validation based on the type name.